### PR TITLE
fix(DB): Tallonkai Swiftroot has incorrect dialogue texts

### LINKED ISSUE #10474: Tallonkai Swiftroot has incorrect dialogue texts
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Original report: https://github.com/chromiecraft/chromiecraft/issues/2909

Tallonkai Swiftroot gives the default "Greetings, <player_name>." text response.

NPC: https://wowgaming.altervista.org/aowow/?npc=3567

Tester:

The issue is confirmed.

### Expected Blizzlike Behaviour

Original report:
Tallonkai Swiftroot should greet the player with dialogue.
https://www.youtube.com/watch?v=YGTw06AC21s&t=34s

Tester:
The correct dialog text should be
`Well met, <name>. It is good to see that <class>s like yourself are taking an active part in protecting the groves.`
(source: https://wowpedia.fandom.com/wiki/Tallonkai_Swiftroot)

### Source

_No response_

### Steps to reproduce the problem

1. `.go creature 49559`
2. Speak with Tallonkai Swiftroot

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/3d6724e448f35ea981d64aff868ce700ab0cbde3

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Add missing gossip for Tallonkai Swiftroot

## Issues Addressed:

- Closes #10474

## Tests Performed:

- Dialog now appears correct (see attach screenshot)
Capture

## How to Test the Changes:

1..go creature 49559
2.Speak with Tallonkai Swiftroot

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
+INSERT INTO `version_db_world` (`sql_rev`)
+VALUES ('1643734354433050200');
+
+UPDATE creature_template
+SET gossip_menu_id=61026
+WHERE entry = 3567;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:6]: Use backticks around the column and table names https://www.azerothcore.org/wiki/sql-standards#general-standards

This line can also be done in one line

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+DELETE
+FROM `npc_text`
+WHERE `ID` = 50032;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:9]: make it one line

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.',
+        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.', 0,
+        0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL,
+        NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1);
+
+DELETE
+FROM `gossip_menu`
+WHERE `MenuID` = 61026
+  AND `TextID` = 50032;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:31]: make this the same line

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+INSERT INTO `gossip_menu` (`MenuID`, `TextID`)
+VALUES (61026, 50032);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:33]:
```suggestion
INSERT INTO `gossip_menu` (`MenuID`, `TextID`) VALUES
(61026, 50032);
```

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
+INSERT INTO `version_db_world` (`sql_rev`)
+VALUES ('1643734354433050200');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:2]: One line

**Review by Kitzunu — CHANGES_REQUESTED** [review]
Yo, thanks for the PR!

There are just some small modifications I'd like you to do to keep with code standards and all that fun stuff.
If you'd like to learn more about the SQL code rules we use, you can check out https://www.azerothcore.org/wiki/sql-standards

**Comment by ZhengPeiRu21** [comment]
Thank you for the feedbacks! I have updated with the suggested changes.

**Review by Kitzunu — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1643734354433050200.sql [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                        `BroadcastTextID5`, `lang5`, `Probability5`, `em5_0`, `em5_1`, `em5_2`, `em5_3`, `em5_4`,
+                        `em5_5`, `text6_0`, `text6_1`, `BroadcastTextID6`, `lang6`, `Probability6`, `em6_0`, `em6_1`,
+                        `em6_2`, `em6_3`, `em6_4`, `em6_5`, `text7_0`, `text7_1`, `BroadcastTextID7`, `lang7`,
+                        `Probability7`, `em7_0`, `em7_1`, `em7_2`, `em7_3`, `em7_4`, `em7_5`, `VerifiedBuild`)
+VALUES (50032,
+        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.',
+        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.', 0,
+        0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL,
+        NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, NULL, NULL, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1643734354433050200.sql:21]:
```suggestion
INSERT INTO `npc_text` (`ID`, `text0_0`, `text0_1`, `BroadcastTextID0`)
VALUES (50032,
        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.',
        'Well met, $N. It is good to see that $cs like yourself are taking an active part in protecting the groves.', 0);
```

**Review by Nefertumm — CHANGES_REQUESTED** [review]

**Review by Nefertumm — APPROVED** [review]

**Comment by Bogir** [comment]
Where did you get `61026 ` number ?
Just last + 1 ?

**Comment by ZhengPeiRu21** [comment]
> Where did you get `61026` number ? Just last + 1 ? `50032` the same?

Yes, I am just choose next free ID for both. It is the highest existing ID + 1, except for a few very high outliers. Please let me know if there is another ID that would be better to use. Thank you!

**Comment by Bogir** [comment]
> > Where did you get `61026` number ? Just last + 1 ? `50032` the same?
> 
> Yes, I am just choose next free ID for both. It is the highest existing ID + 1, except for a few very high outliers. Please let me know if there is another ID that would be better to use. Thank you!

I thought you found the numbers from the original official source.
You acted completely predictably, as the creators of custom dialogs and simple admins of private servers do. But everyone's "last+1" is exactly the one you just used. 
And `DELETE FROM  ...` may cause server error if those Id was allready taken before.

On a good note - we need to document which numbers are reserved for the AС and from which numbers we can start adding texts for "home servers".

This must be documented somehow before merging your PR. 
Let the correct continuous numbering begin with it.

**Comment by ZhengPeiRu21** [comment]
> > > Where did you get `61026` number ? Just last + 1 ? `50032` the same?
> > 
> > 
> > Yes, I am just choose next free ID for both. It is the highest existing ID + 1, except for a few very high outliers. Please let me know if there is another ID that would be better to use. Thank you!
> 
> I thought you found the numbers from the original official source. You acted completely predictably, as the creators of custom dialogs and simple admins of private servers do. But everyone's "last+1" is exactly the one you just used. And `DELETE FROM ...` may cause server error if those Id was allready taken before.
> 
> On a good note - we need to document which numbers are reserved for the AС and from which numbers we can start adding texts for "home servers".
> 
> This must be documented somehow before merging your PR. Let the correct continuous numbering begin with it.
> 
> Or, really, use auto-generated abstract number @last + 1 So this will not cause conflicts with any server's taken numbers

Thank you for kindly explaining. Is there any way to find original ID for this text and menu? I do not know if this is record in sniffs and thought all gossip_menu IDs were arbitrary. Thank you for the correction!

**Comment by Bogir** [comment]
> You are absolutely right that all gossip menu IDs were arbitrary. And very often it causes conflicts on updating live servers. It's time to stop this.

**Comment by Nefertumm** [comment]
The gossip menu id can be sniffed

**Comment by ZhengPeiRu21** [comment]
I have push an update to this branch to also set npcflag to 3 for this NPC, so that gossip will still appear when quest is complete. This is consistent with other NPC that have quest and gossip_menu like Entry 240 (Marshal Dughan).

**Comment by Bogir** [comment]
> I have push an update to this branch to also set npcflag to 3 for this NPC, so that gossip will still appear when quest is complete. This is consistent with other NPC that have quest and gossip_menu like Entry 240 (Marshal Dughan).

I hope you have not only one enUS client for tests?

**Comment by Nefertumm** [comment]
It will not show other language than English because there arent broadcast text for those texts yet. They have to be sniffed.

**Comment by acidmanifesto** [comment]
thank you for hte PR

**Pull request merged**